src/backend/distributed/connection/shared_connection_stats.c

```diff
@@ -121,6 +120,7 @@ citus_remote_connection_stats(PG_FUNCTION_ARGS)
 	TupleDesc tupleDescriptor = NULL;
 
 	CheckCitusVersion(ERROR);
+	RemoveInactiveNodesFromSharedConnections();
```

Comment:
We should be a little careful about calling this function, since it can have weird side effects, e.g. this would wipe out all the entries:

```sql
BEGIN;
DELETE FROM pg_dist_node;
SELECT * FROM citus_remote_connection_stats();
END;
```

Maybe just filter out the inactive nodes in StoreAllRemoteConnectionStats?

Reply:
I have moved the filtering to StoreAllRemoteConnectionStats